### PR TITLE
Upgrade bundled pjproject to 2.15

### PR DESCRIPTION
**Pull request by silentindark**
Fixes: asterisk#1016

UserNote: Bundled pjproject has been upgraded to 2.15. For more
information visit pjproject Github page: https://github.com/pjsip/pjproject/releases/tag/2.15

### CODE REVIEW

**Comment by silentindark** [comment]
cherry-pick-to: 20
cherry-pick-to: 21
cherry-pick-to: 22

**Review by gtjoseph — CHANGES_REQUESTED** [review]
It looks like the patch 0000-configure-ssl-library-path.patch no longer applies cleanly.  I think the changes just need to be shifted down in those files.

**Comment by gtjoseph** [comment]
@silentindark You need to squash your two commits and do a `git push --force`.

**Comment by gtjoseph** [comment]
Four auth tests are now failing because of an assert in pjproject...
```
FAILED: channels/pjsip/basic_calls/incoming/nominal/authed/md5/ident_by_header
FAILED: channels/pjsip/basic_calls/incoming/nominal/authed/md5/ident_by_host
FAILED: channels/pjsip/basic_calls/incoming/nominal/authed/md5/ident_by_user
FAILED: channels/pjsip/basic_calls/incoming/nominal/authed/md5/ident_by_user_drop_options
```
They get fixed by #972 but we need to fix the existing code in this PR.  Stay tuned for more details.  No action needed yet.

**Comment by gtjoseph** [comment]
@silentindark Can you apply this patch and re-push?  It should fix the test failures.

[fixauth.PATCH](https://github.com/user-attachments/files/17996714/fixauth.PATCH)

**Comment by gtjoseph** [comment]
@silentindark What happened?  Was that 1 line patch too much for you? :)

**Comment by silentindark** [comment]
Ups, something went wrong here, but in local repo is fine - https://github.com/asterisk/asterisk/compare/master...silentindark:asterisk-master:master
How I can re-open this PR or push commit to this PR?

**Comment by gtjoseph** [comment]
@silentindark  I reopened it for you but I don't think the patch got applied correctly.  It should have just added 1 line in res_pjsip_authenticator_digest.c but it replaced an existing line instead.

File: res/res_pjsip_authenticator_digest.c [hunk 1]
```diff
@@ -177,7 +177,7 @@ static pj_status_t digest_lookup(pj_pool_t *pool, const pj_str_t *realm,
 		break;
 	case AST_SIP_AUTH_TYPE_MD5:
 		pj_strdup2(pool, &info->data, auth->md5_creds);
-		info->data_type = PJSIP_CRED_DATA_DIGEST;
+		info->algorithm_type = PJSIP_AUTH_ALGORITHM_MD5;
```
Comment (gtjoseph) [inline res/res_pjsip_authenticator_digest.c:180]: The patch didn't apply correctly.  It should have just added the `info->algorithm_type` line leaving the `info->data_type` line alone.

**Review by gtjoseph — CHANGES_REQUESTED** [review]

File: res/res_pjsip_authenticator_digest.c (hunk 1 above)
Comment (silentindark) [reply res/res_pjsip_authenticator_digest.c:180]: Yes, I see. I will fix it.

Comment (silentindark) [reply res/res_pjsip_authenticator_digest.c:180]: Fixed inlocal repo - https://github.com/asterisk/asterisk/compare/master...silentindark:asterisk-master:master

Comment (gtjoseph) [reply res/res_pjsip_authenticator_digest.c:180]: @silentindark  You changed the branch name so I can no longer re-open this pull request.  You have to open a new PR.

**Pull request closed without merging**